tasks/libs/common/check_tools_version.py

```diff
@@ -45,27 +48,41 @@ def custom_golangci_v(v: str) -> str:
     return f"{v}-custom-gcl"
 
 
-def current_golangci_lint_v(ctx: Context) -> str:
+def current_golangci_lint_v(ctx: Context, debug: bool = False) -> str:
     """
     Returns the current user golangci-lint version by running golangci-lint --version
     """
-    cmd = "golangci-lint --version"
-    return ctx.run(cmd, hide=True).stdout.split(' ')[3]
+    debug_flag = "--debug" if debug else ""
+    cmd = f"golangci-lint version {debug_flag}"
+    version_output = ctx.run(cmd, hide=True).stdout
+    return version_output if debug else version_output.split(' ')[3]
 
 
-def check_tools_version(ctx: Context, tools_list: list[str]) -> bool:
+def check_tools_version(ctx: Context, tools_list: list[str], debug: bool = False) -> bool:
     """
     Check that each installed tool in tools_list is the version expected for the repo.
     """
-    is_expected_versions = True
+    should_exit = False
     tools_versions = {
-        'go': {'current_v': current_go_v(ctx), 'expected_v': expected_go_repo_v()},
+        'go': {
+            'current_v': current_go_v(ctx),
+            'expected_v': expected_go_repo_v(),
+            'debug': current_go_v(ctx),
+            'exit_on_error': False,
+            'error_msg': "Warning: If you have linter errors it might be due to version mismatches.",
+        },
         'golangci-lint': {
             'current_v': current_golangci_lint_v(ctx),
             'expected_v': custom_golangci_v(expected_golangci_lint_repo_v(ctx)),
+            'debug': '' if not debug else current_golangci_lint_v(ctx, debug=debug),
+            'exit_on_error': True,
+            'error_msg': "Error: The golanci-lint version you are using is not the correct one. Please run inv -e install-tools to install the correct version.",
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
Would suggest running `inv -e setup` here to install any missing tooling at once

Reply:
Good point, fixed with [d9f11cb](https://github.com/DataDog/datadog-agent/pull/29802/commits/d9f11cb15a85c4171e55cb6d6d4c6a33e01c9799).